presto-spi/src/main/java/com/facebook/presto/spi/type/RowType.java

```diff
@@ -63,6 +63,15 @@ public static RowType anonymous(List<Type> types)
         return new RowType(makeSignature(fields), fields);
     }
 
+    public static RowType withDefaultFieldNames(List<Type> types)
+    {
+        List<Field> fields = new ArrayList<>();
+        for (int i = 0; i < types.size(); i++) {
+            fields.add(new Field(Optional.of("field" + i), types.get(i)));
```

Comment:
I'm not sure what is the convention here. Should it be `field_1`?

Reply:
I'm not sure if there's any convention. It is completely up to us how do we want to generate the filed names. Anyhow those won't be addressed directly in the query, as it is used as an intermediate type for aggregations, and fields are statically accessed in the ordinal way.

Reply:
I am thinking should we make this a static method in, for example `StateCompiler` instead of in broader `RowType`? :)